source/server/active_tcp_socket.cc

```diff
@@ -75,6 +74,29 @@ void ActiveTcpSocket::continueFilterChain(bool success) {
     bool no_error = true;
     if (iter_ == accept_filters_.end()) {
       iter_ = accept_filters_.begin();
+      if (listener_filter_max_read_bytes_ > 0) {
+        listener_filter_buffer_ = std::make_unique<Network::ListenerFilterBufferImpl>(
+            socket_->ioHandle(), listener_.dispatcher(),
+            [this](bool error) {
+              socket_->ioHandle().close();
+              error ? listener_.stats_.downstream_peek_error_.inc()
+                    : listener_.stats_.downstream_peek_remote_close_.inc();
+              continueFilterChain(false);
+            },
+            [this](Network::ListenerFilterBuffer& filter_buffer) {
+              ASSERT((*iter_)->maxReadBytes() > 0);
+              Network::FilterStatus status = (*iter_)->onData(filter_buffer);
+              if (status == Network::FilterStatus::StopIteration) {
+                if (socket_->ioHandle().isOpen()) {
+                  // There is no more data when the buffer reaches the max read bytes.
+                  ASSERT(filter_buffer.rawSlice().len_ < listener_filter_max_read_bytes_);
```

Comment:
Is this validating that the listener filter isn't waiting for more data when we've given it all the data it requested? If so, I'd reword the comment to `The listener filter should not wait for more data when it has already received all the data it requested.` or similar.

Reply:
yes, exactly what you say, I will correct the comments.